### ISSUE TITLE
MRSV prevents `jpeg-decoder` to compile when `test-dev-graph` feature is enabled

### ISSUE THREAD

**Issue by CPerezz** [zcash/halo2#719]
I was working on a PR locally when I ran clippy with `cargo clippy --all-features --all-targets -- -D warnings`. This led to the error: 

```
error: package `jpeg-decoder v0.3.0` cannot be built because it requires rustc 1.61.0 or newer, while the currently active rustc version is 1.56.1
```

A quick search with `cargo tree --all-features --invert=jpeg-decoder ` lead to:
```
jpeg-decoder v0.3.0
└── image v0.24.5
    └── plotters-bitmap v0.3.2
        └── plotters v0.3.4
            ├── criterion v0.3.6
            │   └── pprof v0.8.0
            │       [dev-dependencies]
            │       └── halo2_gadgets v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_gadgets)
            │   [dev-dependencies]
            │   ├── halo2_gadgets v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_gadgets)
            │   └── halo2_proofs v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_proofs)
            │       ├── halo2 v0.1.0-beta.2 (/home/kr0/Desktop/HDD/dev/halo2/halo2)
            │       └── halo2_gadgets v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_gadgets)
            ├── halo2_gadgets v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_gadgets)
            └── halo2_proofs v0.2.0 (/home/kr0/Desktop/HDD/dev/halo2/halo2_proofs) (*)
```
Which leaves clear that the feature that causes the issue is `test-dev-graph` which invokes `"plotters/bitmap_backend"`.

Should we update the MSRV? This would also allow to maybe get GATs and some other features avaliable if at any point are needed.

NOTE: This will also prevent the CI to compile since Clippy uses `--all-features` and therefore will try to use `test-dev-graph` too.

**Comment by str4d** [comment]
The MSRV applies to using the library in applications, not necessarily to the development tooling. But this is pretty annoying for developer UX. It would be nice if we could have separate MSRVs for each use case.

In the interim you can use `cargo +stable test` etc.

**Comment by CPerezz** [comment]
Oh, I don't have issues with the MSRV per-se. But we execute clippy in the tests with `--all-features`. And that will cause that no tests will pass.

**Comment by str4d** [comment]
I think what we should do is only run tests on crate-internal feature flags against `stable`, and run MSRV-level tests and lints excluding these features.

**Comment by CPerezz** [comment]
Sounds good! Should I submit a PR or you'll take care?

**Comment by str4d** [comment]
Hmm, it turns out we need to raise our MSRV anyway, because `constant_time_eq` as of 7 months ago requires Rust 1.59 because it uses inline assembly. That is a dependency of `blake2b_simd`, so we need to bump our MSRV. I have no idea why CI has been passing until now; we don't check in a `Cargo.lock` so we should have seen this issue months ago.

**Comment by str4d** [comment]
Aha, `blake2b_simd` released 1.0.1 which migrated from `constant_time_eq 0.1` to `constant_time_eq 0.2`.
